eel/edge.py

```diff
@@ -1,20 +1,48 @@
-import platform
+import os
 import subprocess as sps
 import sys
 from typing import List
 
 from eel.types import OptionsDictT
 
-name: str = 'Edge'
+name: str = "Edge"
 
 
-def run(_path: str, options: OptionsDictT, start_urls: List[str]) -> None:
-    cmd = 'start microsoft-edge:{}'.format(start_urls[0])
-    sps.Popen(cmd, stdout=sys.stdout, stderr=sys.stderr, stdin=sps.PIPE, shell=True)
+def run(path: str, options: OptionsDictT, start_urls: List[str]) -> None:
+    if path != "edge_legacy":
+        if options["app_mode"]:
+            for url in start_urls:
+                sps.Popen([path, "--app=%s" % url] + options["cmdline_args"], stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+        else:
+            args = options["cmdline_args"] + start_urls
+            sps.Popen([path, "--new-window"] + args, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+    else:
+        cmd = "start microsoft-edge:{}".format(start_urls[0])
+        sps.Popen(cmd, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE, shell=True)
 
 
 def find_path() -> bool:
-    if platform.system() == 'Windows':
-        return True
+    if sys.platform in ["win32", "win64"]:
+        return _find_edge_win()
+    else:
+        return None
 
-    return False
+
+def _find_edge_win():
```

Comment:
This function needs a return type annotation, `str` is best if you're confident it will always return a string.